### PR TITLE
Feat/306 orchard modal

### PR DESCRIPTION
**Pull request by ArthurEncr**
# Description

Display warning message modal when orchard ID is added or deleted, under the circumstance that the parent tree table data object is not empty. (The "change" modal is done but not implemented as the design is not done yet.)

Fixes # [nr-spar #306](https://github.com/bcgov/nr-spar/issues/306)

## How to test?
Go to the Seedlot Registration in step 4 (orchard), try to add/delete an orchard and no modal should appear. After that, type an orchard and press "enter" (for example: "405"), then go to step 5 and wait for the table to load, go back to step 4 and try to add or delete an orchard, the modal should appear.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-343-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-343-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-343-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

### CODE REVIEW

**Comment by craigyu** [comment]
I've made some changes [here] to prepare for the merge with #332 (https://github.com/bcgov/nr-spar/pull/343/commits/2eddbdf51a9f38a55d09cab69abff7b18e35503b)

The key takeaway is try to have a stronger type definition. Avoid the usage of `[key: string]: someType` when possible.

File: frontend/src/components/SeedlotRegistrationSteps/OrchardStep/OrchardModal/styles.scss [hunk 1]
```diff
@@ -0,0 +1,9 @@
+@use '@carbon/type';
+@use '../../../../theme/variables.scss' as vars;
```
Comment (mgaseta) [inline frontend/src/components/SeedlotRegistrationSteps/OrchardStep/OrchardModal/styles.scss:2]: This variable is not being used, you can remove it. :)

File: frontend/src/components/SeedlotRegistrationSteps/OrchardStep/OrchardModal/styles.scss [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+
```
Comment (mgaseta) [inline frontend/src/components/SeedlotRegistrationSteps/OrchardStep/OrchardModal/styles.scss:4]: Remove extra blank line.

**Review by craigyu — CHANGES_REQUESTED** [review]
The buttons are too big:
- Expected
    - image
- Actual
    - image

File: frontend/src/components/SeedlotRegistrationSteps/OrchardStep/OrchardModal/styles.scss (hunk 1 above)
Comment (craigyu) [reply frontend/src/components/SeedlotRegistrationSteps/OrchardStep/OrchardModal/styles.scss]: [fixed here](https://github.com/bcgov/nr-spar/pull/343/commits/e1dcf8de2ed40cb2d1c39dd01f06ccee123c7416)

File: frontend/src/components/SeedlotRegistrationSteps/OrchardStep/OrchardModal/styles.scss (hunk 2 above)
Comment (craigyu) [reply frontend/src/components/SeedlotRegistrationSteps/OrchardStep/OrchardModal/styles.scss:2]: [fixed here](https://github.com/bcgov/nr-spar/pull/343/commits/e1dcf8de2ed40cb2d1c39dd01f06ccee123c7416)

**Comment by craigyu** [comment]
> The buttons are too big:
> 
> * Expected
>   
>   * image
> * Actual
>   
>   * image

[fix](https://github.com/bcgov/nr-spar/pull/343/commits/8e8c7daca454815cecc6dedf352e6fd05e405620)

**Review by craigyu — APPROVED** [review]

**Review by mgaseta — CHANGES_REQUESTED** [review]
Nice job! 
Only one change in relation to the design, the label in the modal title should be smaller
- Current:
Screenshot from 2023-07-19 14-31-54
- Expected:
Screenshot from 2023-07-19 14-33-36

We have the correct rule to apply the label style, but it's being overwritten by another rule:
Screenshot from 2023-07-19 14-35-25

**Review by mgaseta — APPROVED** [review]
LGTM! :rocket: :tada:

**Pull request merged**